tests/Graphics/Implicit/Test/Instances.hs

```diff
@@ -182,17 +234,65 @@ instance Observe () Double Double
 a =~= b = Test.QuickCheck.property $ \test -> observe test a Test.QuickCheck.=== observe test b
 infix 4 =~=
 
+makeTestResult
+  :: forall obj f a
+   . ( RealFloat a
+     , Object obj f a
+     , Quantizable a
+     , f ~ Space obj
+     )
+  => obj
+  -> Space obj a
+  -> TestResult obj a
+makeTestResult obj sampleAt =
+  let
+    fun = getImplicit obj
+    sampledVal = quantize epsilon $ fun sampleAt
+  in
+    TestResult
+      { trInsidedness = insidedness sampledVal
+      , trSampledValue = sampledVal
+      , trSampledObject = obj
+      , trSampledAt = sampleAt
+      }
+
 ------------------------------------------------------------------------------
 -- | Two 'SymbolicObj2's are the same if their 'getImplicit' functions agree at
 -- all points (up to an error term of 'epsilon')
-instance Observe (ℝ2, ()) Insidedness SymbolicObj2 where
-  observe p = insidedness . observe p . getImplicit
+instance Observe (ℝ2, ()) (TestResult SymbolicObj2 Double) SymbolicObj2 where
+  observe (sampledAt, _) obj = makeTestResult obj sampledAt
 
 ------------------------------------------------------------------------------
 -- | Two 'SymbolicObj3's are the same if their 'getImplicit' functions agree at
 -- all points (up to an error term of 'epsilon')
-instance Observe (ℝ3, ()) Insidedness SymbolicObj3 where
-  observe p = insidedness . observe p . getImplicit
+instance Observe (ℝ3, ()) (TestResult SymbolicObj3 Double) SymbolicObj3 where
+  observe (sampledAt, _) obj = makeTestResult obj sampledAt
+
+------------------------------------------------------------------------------
+-- | The number of decimal points we need to agree to assume two 'Double's are
+-- equal.
```

Comment:
excuse me while i hurl. :)

Reply:
I think that the reason it wasn't required was that `Arbitrary Double / Gen Double` wasn't generating small enough or close to zero things and transforms. Before, there was `epsilon = 5` and currently even `epsilon = 16` works quite well but eventually, with large enough `--qc-max-success` it breaks. This seems to hold up to `--qc-max-success=10M`